tests/Unit/PointwiseFunctions/MathFunctions/TestMathHelpers.hpp

```diff
@@ -0,0 +1,30 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Contains helper functions for testing math functions
+
+#pragma once
+
+#include <random>
+
+#include "tests/Unit/TestHelpers.hpp"
+
+///
+/// \ingroup TestHelpers
+/// \brief Serializes and deserializes the MathFunction `function` and checks
+/// that the resulting function gives the same results when applied to some
+/// random input values
+template <typename Function>
+void test_pup_function(const Function& function) {
```

Comment:
Should this specify dimensionality? Right now we only have 1D `MathFunction` but I thought there was discussion of possibly needing 2D and 3D.

Reply:
Currently a TODO, the version to work with 2 or 3D functions would have to be overloaded anyway since they would take arrays of doubles instead of just doubles. I think it'd make more sense to figure out the correct way to handle the dimensionality in the pull request that adds 2D and 3D functions.

Reply:
Sounds good to me